test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageFileServiceTests.cs

```diff
@@ -153,6 +193,45 @@ public async Task GetPackageIconAsync_EmbeddedFromFallbackFolder_DoesNotLockFile
             }
         }
 
+        [Fact]
+        public async Task GetPackageIconAsync_NonEmbedded_GetStream_DoesNotLockFile()
+        {
+            // Arrange
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                string fileContents = "testFileContents";
+                string filePath = Path.Combine(testDirectory.Path, "testFile.txt");
+
+                File.WriteAllText(filePath, fileContents);
+
+                var packageFileService = new NuGetPackageFileService(
+                        default(ServiceActivationOptions),
+                        Mock.Of<IServiceBroker>(),
+                        new AuthorizationServiceClient(Mock.Of<IAuthorizationService>()),
+                        _telemetryProvider.Object);
+                var packageIdentity = new PackageIdentity(nameof(GetPackageIconAsync_NonEmbedded_GetStream_DoesNotLockFile), NuGetVersion.Parse("1.0.0"));
+
+                // Do not add a fragment which is typically required for the embedded icon.
+                // Note: file:// is required for Uri to parse the Fragment (possibly a Uri bug).
+                var uri = new Uri("file://" + filePath);
+                NuGetPackageFileService.AddIconToCache(packageIdentity, uri);
+                FileStream fileNotLocked = File.Open(uri.LocalPath, FileMode.Open, FileAccess.ReadWrite, FileShare.Read);
+
+                // Act
+                // System.IO.IOException would occur in the Act if we're locking the icon file.
+                Stream iconStream = await packageFileService.GetPackageIconAsync(packageIdentity, CancellationToken.None);
+                fileNotLocked.Close();
+                FileStream fileNotLocked2 = File.Open(uri.LocalPath, FileMode.Open, FileAccess.ReadWrite, FileShare.Read);
```

Comment:
multiple streams not getting disposed here, so like my other comment, I expect that the test directory is not getting delete when the test finishes.

Reply:
Yes - I had noticed a cluttered test folder, but forgot to go back and cleanup the Streams. done now